File: src/387. First Unique Character in a String.js
/**
 * @param {string} s
 * @return {number}
 */

export function firstUniqChar(s) {
  const map = new Map()
  for (let i = 0; i < s.length; i++) {
    if (map.has(s[i])) map.set(s[i], -1)
    else map.set(s[i], i)
  }
  const values = [...map.values()]
  return values.find((value) => value >= 0) ?? -1
}

File: src/387. First Unique Character in a String.cpp
class Solution {
    public:
        int firstUniqChar(string s) {
            int n = s.size();
            int counts[26] = {
                0
            };
            for (int i = 0; i < n; i++) {
                counts[s[i] - 'a']++;
            }

            for (int i = 0; i < n; i++) {
                if (counts[s[i] - 'a'] == 1) {
                    return i;
                }
            }
            return -1;
        }
};